tests/test_downloader.c

```diff
@@ -290,6 +290,100 @@ START_TEST(test_downloader_three_files_with_error)
 }
 END_TEST
 
+START_TEST(test_downloader_checksum)
+{
+    #define NUM_CHECKSUM_TESTS 2
```

Comment:
I'm not used to doing that with structs.  Seems like a good idea, see second commit.

(And apologies for rebasing the first commit, I'm not sure what went wrong but I pushed a version that failed to build due to parameter mismatch).